src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -207,55 +201,50 @@ grpc_channel_credentials* grpc_google_default_credentials_create(void) {
   if (err == GRPC_ERROR_NONE) goto end;
   error = grpc_error_add_child(error, err);
 
+  gpr_mu_lock(&g_state_mu);
```

Comment:
It is not related to the bug, but if we have another use case that fetches the token from metadata server (instead of env var),  we may have a same issue here.